src/components/BookingModal/BookingModalForm/index.js

```diff
@@ -24,29 +25,32 @@ const BookingModalForm = props => {
     booking,
     isSameDay,
   } = props;
-
+  const { eventStatus: {eventStatusId} } = booking;
+  const isEventCancelled = eventStatusId === holidayStatus.CANCELLED;
+  const rejectionMessages = booking.messages && !isEventCancelled;
   const { eventTypeId } = formData;
   const buttonTextValue =
   `${eventTypeId !== eventTypes.ANNUAL_LEAVE ? '' : bookingDuration === 0.5 ? 'Half' : bookingDuration}
   ${eventTypeId !== eventTypes.ANNUAL_LEAVE ? 'WFH' : bookingDuration > 1 ? 'Days' : 'Day'}`;
 
   const createCtas = () => {
+    const isButtonDisabled = isEventCancelled ? isEventCancelled : !formIsValid;
     if (isEventBeingUpdated) {
       return [
         {
-          label: `Update to ${
+          label: isEventCancelled ? 'Cancelled' : rejectionMessages ? 'Submit' : `Update to ${
             bookingDuration === 0.5 ? 'Half' : bookingDuration
           } ${bookingDuration > 1 ? 'Days' : 'Day'}`,
           event: updateEvent,
-          disabled: !formIsValid,
+          disabled: isButtonDisabled,
```

Comment:
This will likely cause conflicts with the refactor branch because we're now driving this using the `isValidBooking` function in the container. Again, I recommend a merge.